packages/mermaid/src/diagrams/xychart/xychartDb.ts

```diff
@@ -0,0 +1,302 @@
+  text: string;
+}
+
+function textSanitizer(text: string) {
+  return sanitizeText(text.trim(), config);
+}
+
+function parseDirective(statement: string, context: string, type: string) {
+  // @ts-ignore: TODO Fix ts errors
+  mermaidAPI.parseDirective(this, statement, context, type);
+}
+
+function setOrientation(oriantation: string) {
+  if (oriantation === 'horizontal') {
+    xyChartConfig.chartOrientation = 'horizontal';
+  } else {
+    xyChartConfig.chartOrientation = 'vertical';
+  }
+}
+function setXAxisTitle(title: NormalTextType) {
+  xyChartData.xAxis.title = textSanitizer(title.text);
+}
+function setXAxisRangeData(min: number, max: number) {
+  xyChartData.xAxis = { type: 'linear', title: xyChartData.xAxis.title, min, max };
+  hasSetXAxis = true;
+}
+function setXAxisBand(categories: NormalTextType[]) {
+  xyChartData.xAxis = {
+    type: 'band',
+    title: xyChartData.xAxis.title,
+    categories: categories.map((c) => textSanitizer(c.text)),
+  };
+  hasSetXAxis = true;
+}
+function setYAxisTitle(title: NormalTextType) {
+  xyChartData.yAxis.title = textSanitizer(title.text);
+}
+function setYAxisRangeData(min: number, max: number) {
+  xyChartData.yAxis = { type: 'linear', title: xyChartData.yAxis.title, min, max };
+  hasSetYAxis = true;
+}
+
+// this function does not set `hasSetYAxis` as there can be multiple data so we should calculate the range accordingly
+function setYAxisRangeFromPlotData(data: number[]) {
+  const minValue = Math.min(...data);
+  const maxValue = Math.max(...data);
+  const prevMinValue = isLinearAxisData(xyChartData.yAxis) ? xyChartData.yAxis.min : Infinity;
+  const prevMaxValue = isLinearAxisData(xyChartData.yAxis) ? xyChartData.yAxis.max : -Infinity;
+  xyChartData.yAxis = {
+    type: 'linear',
+    title: xyChartData.yAxis.title,
+    min: Math.min(prevMinValue, minValue),
+    max: Math.max(prevMaxValue, maxValue),
+  };
+}
+
+function transformDataWithOutCategory(data: number[]): SimplePlotDataType {
+  let retData: SimplePlotDataType = [];
+  if (data.length === 0) {
+    return retData;
+  }
+  if (!hasSetXAxis) {
+    const prevMinValue = isLinearAxisData(xyChartData.xAxis) ? xyChartData.xAxis.min : Infinity;
+    const prevMaxValue = isLinearAxisData(xyChartData.xAxis) ? xyChartData.xAxis.max : -Infinity;
+    setXAxisRangeData(Math.min(prevMinValue, 1), Math.max(prevMaxValue, data.length));
+  }
+  if (!hasSetYAxis) {
+    setYAxisRangeFromPlotData(data);
+  }
+
+  if (isBandAxisData(xyChartData.xAxis)) {
+    retData = xyChartData.xAxis.categories.map((c, i) => [c, data[i]]);
+  }
+
+  if (isLinearAxisData(xyChartData.xAxis)) {
+    const min = xyChartData.xAxis.min;
+    const max = xyChartData.xAxis.max;
+    const step = (max - min + 1) / data.length;
+    const categories: string[] = [];
+    for (let i = min; i <= max; i += step) {
+      categories.push(`${i}`);
+    }
+    retData = categories.map((c, i) => [c, data[i]]);
+  }
+
+  return retData;
+}
+
+let plotIndex = 0;
```

Comment:
Move all the file level variables to the top, and inside an object. 
Then, we can clear all the global variables with a single statement. Otherwise, when someone is making changes later, they might miss out stuff that's scattered throughout the file.

Reply:
Done